reaper.go

```diff
@@ -87,55 +87,53 @@ func lookUpReaperContainer(ctx context.Context, sessionID string) (*DockerContai
 	exp.MaxInterval = 5.0 * time.Second  // max interval between attempts
 	exp.MaxElapsedTime = 1 * time.Minute // max time to keep trying
 
-	var reaperContainer *DockerContainer
-	err = backoff.Retry(func() error {
-		args := []filters.KeyValuePair{
+	opts := container.ListOptions{
+		All: true,
+		Filters: filters.NewArgs(
 			filters.Arg("label", fmt.Sprintf("%s=%s", core.LabelSessionID, sessionID)),
 			filters.Arg("label", fmt.Sprintf("%s=%t", core.LabelReaper, true)),
 			filters.Arg("label", fmt.Sprintf("%s=%t", core.LabelRyuk, true)),
 			filters.Arg("name", reaperContainerNameFromSessionID(sessionID)),
-		}
-
-		resp, err := dockerClient.ContainerList(ctx, container.ListOptions{
-			All:     true,
-			Filters: filters.NewArgs(args...),
-		})
-		if err != nil {
-			return err
-		}
-
-		if len(resp) == 0 {
-			// reaper container not found in the running state: do not look for it again
-			return nil
-		}
+		),
+	}
 
-		if len(resp) > 1 {
-			return fmt.Errorf("not possible to have multiple reaper containers found for session ID %s", sessionID)
-		}
+	return backoff.RetryNotifyWithData(
+		func() (*DockerContainer, error) {
+			resp, err := dockerClient.ContainerList(ctx, opts)
+			if err != nil {
+				return nil, err
+			}
 
-		r, err := containerFromDockerResponse(ctx, resp[0])
-		if err != nil {
-			return err
-		}
+			if len(resp) == 0 {
+				// reaper container not found in the running state: do not look for it again
+				return nil, nil
+			}
 
-		reaperContainer = r
```

Comment:
We probably missed updating the reaperContainer singleton instance

Reply:
This was setting the variable on line 90 that was then returned.

Reply:
Ah sorry I see it now that it's a RetryNotifyWithData 🤦 Sorry!